src/internal/observe/alias.go

```diff
@@ -0,0 +1,9 @@
+package observe
+
+// IDName is used to pass around a tuple representing
+// some id (such as a resource owner), and some display
+// name that end users would find more easy to consume.
+type IDName struct {
```

Comment:
I feel like from a scalability standpoint (i.e. how far can we push the design for different types/situations) we may be better off using an interface instead of a concrete struct for this and for the Selector/Name struct defined in the CLI layer.

Together, these structs are trying to represent:
* the human readable name for something,
* the ID of a resource so we can get it with Graph
* the other selectors for the resource

By making a set of concrete structs, we assume we can always squeeze all of the above down to what the structs define and we don't leave ourselves an easy way forward if they don't. It also ties generating all of that information to struct creation time (i.e. the struct needs to have those fields set immediately otherwise the struct ceases to be useful) and forces the code that creates the struct to know exactly how to generate the data. The latter especially spreads type-specific logic around the codebase instead of centralizing it (i.e. if both GC and CLI layers need some of the IDName stuff, they both need to know how to make it). We also have a bit of what feels like an anti-pattern where we make a struct and then continue to individually access and pass around separate struct values (i.e. call to `repo.NewBackup`). That spreads the handling of all of these separate things to other components as well

If we use an interface and have implementations for the interface then we can
* delay generating ID/Name values until the functions are called
* centralize the code that needs to know how to generate various values
* fine-tune each data type to have different outputs easily because there's only one place to change things

Just to spitball an interface design here (this need not necessarily be defined in this file, but since I'm already commenting here...)
```go
// idk, can't name things this morning
type ResourceNamer interface {
  // DisplayName returns a human-readable name for the resource
  DisplayName() string
  // ResourceID returns the external service ID for the resource
  ResourceID() string
}

type DisplaySelector interface {
  ResourceNamer
  // Selector returns the selector being used for this operation
  Selector() Selector
}
```

The observe package could use the `ResourceNamer` interface (so we don't give it access to more functions than it needs) and other things could take the `DisplaySelector` interface. That way we continue to pack the data together, but allow accessing all of it through well-defined interfaces

Reply:
Overall: agree that this is the wrong pattern here, I'm going to work on that.  Beyond that, general disagreements on the particulars.  Primarily that this is an attempt to represent selectors, and that interfaces are an effective solution to this situation.

What's needed is a display name for the resource owner, and I overreached in turning this into a struct.  Simplifying the implementation to the needs at hand will be the best way forward.

Reply:
@ashmrtn could you kindly check out the new iteration?